packages/PHPStanStaticTypeMapper/TypeMapper/HasOffsetTypeMapper.php

```diff
@@ -38,7 +37,7 @@ public function mapToPHPStanPhpDocTypeNode(Type $type, TypeKind $typeKind): Type
     /**
      * @param HasOffsetType $type
      */
-    public function mapToPhpParserNode(Type $type, TypeKind $typeKind): ?Node
+    public function mapToPhpParserNode(Type $type, TypeKind $typeKind): never
```

Comment:
Is this lega, since the return type of ?Node was part of TypeMapperInterface contract/interface?

Reply:
It seems cause error on scoping https://github.com/rectorphp/rector-src/runs/4418027866?check_suite_focus=true#step:23:7

```
Run cd remote-repository && bin/rector list --ansi && cd ..
  cd remote-repository && bin/rector list --ansi && cd ..
  shell: /usr/bin/bash -e {0}
  env:
    COMPOSER_ROOT_VERSION: dev-main
PHP Fatal error:  Declaration of Rector\PHPStanStaticTypeMapper\TypeMapper\HasOffsetTypeMapper::mapToPhpParserNode($type, $typeKind) must be compatible with Rector\PHPStanStaticTypeMapper\Contract\TypeMapperInterface::mapToPhpParserNode($type, $typeKind): ?PhpParser\Node in /home/runner/work/rector-src/rector-src/remote-repository/packages/PHPStanStaticTypeMapper/TypeMapper/HasOffsetTypeMapper.php on line 17
Fatal error: Declaration of Rector\PHPStanStaticTypeMapper\TypeMapper\HasOffsetTypeMapper::mapToPhpParserNode($type, $typeKind) must be compatible with Rector\PHPStanStaticTypeMapper\Contract\TypeMapperInterface::mapToPhpParserNode($type, $typeKind): ?PhpParser\Node in /home/runner/work/rector-src/rector-src/remote-repository/packages/PHPStanStaticTypeMapper/TypeMapper/HasOffsetTypeMapper.php on line 17
Error: Process completed with exit code 255
```

I will check.

Reply:
Fixed at https://github.com/rectorphp/rector-src/pull/1390